Mage.Sets/src/mage/cards/b/BraidsArisenNightmare.java

```diff
@@ -0,0 +1,148 @@
+package mage.cards.b;
+
+import java.util.HashSet;
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfYourEndStepTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.Predicate;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetControlledPermanent;
+
+/**
+ *
+ * @author weirddan455
+ */
+public final class BraidsArisenNightmare extends CardImpl {
+
+    public BraidsArisenNightmare(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{B}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.NIGHTMARE);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // At the beginning of your end step, you may sacrifice an artifact, creature, enchantment, land, or planeswalker.
+        // If you do, each opponent may sacrifice a permanent that shares a card type with it.
+        // For each opponent who doesn't, that player loses 2 life and you draw a card.
+        this.addAbility(new BeginningOfYourEndStepTriggeredAbility(new BraidsArisenNightmareEffect(), true));
```

Comment:
Could this be simplified with a DoIfCostPaid?

Reply:
I looked at that but it needs some way to give the information about what was sacrificed to the effect (so opponents will have to sacrifice the same type).  Unless you know of a way to do that that's cleaner than what I wrote?

Reply:
DoIfCostPaid would probably have to be changed, but I don't think it's worth it since I didn't find many other cards which need that info.